tests/service_controller_test.go

```diff
@@ -0,0 +1,48 @@
+package tests
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	v1 "k8s.io/api/core/v1"
+	"kubevirt.io/ssp-operator/controllers"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func getSspMetricsService() (*v1.Service, error) {
+	service := controllers.ServiceObject(strategy.GetSSPDeploymentNameSpace())
+	err := apiClient.Get(ctx, client.ObjectKeyFromObject(service), service)
+	return service, err
+}
+
+var _ = Describe("Service Controller", func() {
+	var (
+		serviceErr error
+		service    *v1.Service
+	)
+	BeforeEach(func() {
+		waitUntilDeployed()
+
+		service, serviceErr = getSspMetricsService()
+	})
+
+	Context("creation", func() {
+		It("[test_id: ???] Should create ssp-operator-metrics service", func() {
+			if serviceErr != nil {
+				Fail("Failed to get ssp-operator-metrics service")
+				return
+			}
+		})
+
+		It("[test_id: ???] Should re-create ssp-operator-metrics service if deleted", func() {
+			Expect(apiClient.Delete(ctx, service)).To(Succeed())
+			Eventually(func() error {
+				err := apiClient.Get(ctx, client.ObjectKeyFromObject(service), service)
```

Comment:
This should also test that the new `service` has different `UID` that the previous service, so we are sure that it was deleted and recreated.